src/sql/parts/objectExplorer/viewlet/serverTreeView.ts

```diff
@@ -227,16 +227,17 @@ export class ServerTreeView {
 		}
 	}
 
-	public deleteObjectExplorerNodeAndRefreshTree(connection: IConnectionProfile): void {
+	public deleteObjectExplorerNodeAndRefreshTree(connection: IConnectionProfile): Thenable<void> {
 		if (connection) {
 			var conn = this.getConnectionInTreeInput(connection.id);
 			if (conn) {
-				this._objectExplorerService.deleteObjectExplorerNode(conn).then(() => {
+				return this._objectExplorerService.deleteObjectExplorerNode(conn).then(() => {
 					this._tree.collapse(conn);
 					this._tree.refresh(conn);
 				});
 			}
 		}
+		return new Promise<void>(()=>{});
```

Comment:
you could also use `return Promise.resolve()` since it's simpler

Reply:
thanks, I was looking for that syntax